ctapipe_io_magic/__init__.py

```diff
@@ -1037,12 +1120,12 @@ def simulation_config(self):
 
     @property
     def datalevels(self):
-        return (self.datalevel, )
+        return (self.datalevel,)
 
     @property
     def obs_ids(self):
         # ToCheck: will this be compatible in the future, e.g. with merged MC files
-        return list(self.observation_blocks)
+        return self.run_id
```

Comment:
`obs_ids` should return a list (see https://github.com/cta-observatory/ctapipe/blob/main/ctapipe/io/eventsource.py#L277), but `self.run_id` is a simple integer. So it should be cast to a list. Not sure how tests could have passed. If you cannot test because the files are at PIC where you cannot access them, let me know and I will put them somewhere were you can grab them

Reply:
For reference:
```bash
FAILED ctapipe_io_magic/tests/test_magic_event_source.py::test_loop[dataset4] - TypeError: 'int' object is not subscriptable
FAILED ctapipe_io_magic/tests/test_magic_event_source.py::test_loop[dataset5] - TypeError: 'int' object is not subscriptable
FAILED ctapipe_io_magic/tests/test_magic_event_source.py::test_loop[dataset6] - TypeError: 'int' object is not subscriptable
FAILED ctapipe_io_magic/tests/test_magic_event_source.py::test_loop[dataset7] - TypeError: 'int' object is not subscriptable
FAILED ctapipe_io_magic/tests/test_magic_event_source.py::test_subarray_multiple_runs - assert [824318] == 824318
FAILED ctapipe_io_magic/tests/test_magic_event_source.py::test_that_event_is_not_modified_after_loop[dataset4] - TypeError: 'int' object is not subscriptable
FAILED ctapipe_io_magic/tests/test_magic_event_source.py::test_that_event_is_not_modified_after_loop[dataset5] - TypeError: 'int' object is not subscriptable
FAILED ctapipe_io_magic/tests/test_magic_event_source.py::test_that_event_is_not_modified_after_loop[dataset6] - TypeError: 'int' object is not subscriptable
FAILED ctapipe_io_magic/tests/test_magic_event_source.py::test_that_event_is_not_modified_after_loop[dataset7] - TypeError: 'int' object is not subscriptable
FAILED ctapipe_io_magic/tests/test_stage1.py::test_stage1_multiple_runs - TypeError: 'int' object is not iterable
FAILED ctapipe_io_magic/tests/test_stage1.py::test_stage1_single_run - TypeError: 'int' object is not iterable
```

Reply:
> `obs_ids` should return a list (see https://github.com/cta-observatory/ctapipe/blob/main/ctapipe/io/eventsource.py#L277), but `self.run_id` is a simple integer. So it should be cast to a list. Not sure how tests could have passed. If you cannot test because the files are at PIC where you cannot access them, let me know and I will put them somewhere were you can grab them

`list(self.run_id)` does not work, since `list` requires the argument to be an iterable, so the previous solution `list(self.observation_blocks)` or `list(self.observation_blocks.keys())` should be used.